Sprint-3/4-stretch-investigate/password-validator.js

```diff
@@ -1,6 +1,36 @@
-function passwordValidator(password) {
-    return password.length < 5 ? false : true
-}
+function passwordValidator(password,previousPasswords) {
+      // Check if the password is at least 5 characters long
+  if (password.length < 5) {
+    return false; // Invalid if password is too short
+  }
+
+  // Check if password contains at least one uppercase letter
+  if (!/[A-Z]/.test(password)) {
+    return false; // Invalid if no uppercase letter
+  }
+
+  // Check if password contains at least one lowercase letter
+  if (!/[a-z]/.test(password)) {
+    return false; // Invalid if no lowercase letter
+  }
+
+  // Check if password contains at least one number
+  if (!/[0-9]/.test(password)) {
+    return false; // Invalid if no number
+  }
 
+  // Check if password contains at least one special character
+  if (!/[!#$%.&*]/.test(password)) {
```

Comment:
This will do fine for just now - there are a very large number of possible "special characters" (a term usually meaning characters that are not letters or numbers) but this is a few of them.